### PR TITLE
Add variable display size suport + add support for 4" display module st7796s. Speedup SPI bus

### PR DESCRIPTION
**Pull request by DiSlord**
Replace __REV16 as inline asm instruction
Remove delay in send command
This all allow increase display draw speed (and reduce code size)

### CODE REVIEW

File: ili9341.cpp [hunk 1]
```diff
@@ -261,13 +261,22 @@ ili9341_init(void)
...
 }
-
+#if 0
 static inline uint32_t __REV16(uint32_t rev)
 {
 	uint32_t a = __builtin_bswap16(rev >> 16);
 	uint32_t b = __builtin_bswap16(rev & 0xFFFF);
 	return (a << 16) | (b);
 }
+#else
```
Comment (wutje) [inline ili9341.cpp:271]: Just drop my ulgy code and use only use the inline asm. Not need to keep this around.

**Comment by DiSlord** [comment]
Ok, i remove it on next commits. 

__REV16 (rev16) is standart Cortex instruction supported by all models, and allow save some flash size

You know why for fill in
`void ili9341_fill(int x, int y, int w, int h, uint16_t color)`
used ili9341_spi_buffer as temp data?

More easy and faster use direct dma transfer single color data vs memory increment = false

File: ili9341.cpp [hunk 2]
```diff
@@ -261,13 +261,22 @@ ili9341_init(void)
...
+// Reverses the byte order within each halfword of a word. For example, 0x12345678 becomes 0x34127856.
+static inline uint32_t __REV16(uint32_t value)
+{
+  uint32_t result;
+  __asm volatile("rev16 %0, %1" : "=r" (result) : "r" (value));
```
Comment (gabriel-tenma-white) [inline ili9341.cpp:276]: I suggest moving inline assembly stuff to its own file, maybe a utility.hpp that contains these inline functions. In the future I think there are going to be risc-v based instruments.

Comment (wutje) [reply ili9341.cpp:276]: If multiple cpu architectures need to be supported then we should keep this. Utility.hpp seems fine.

Just checking: there is no direct built in for this?

**Comment by DiSlord** [comment]
Add fonts like in NanoVNA-H
Update render plot functions
Set some display draw constants as define

Screen render speed should be faster

Try test, but need continue fix ui.cpp for replace size hardcoded variables

**Comment by DiSlord** [comment]
Update big font
Many size definition fixes

**Pull request merged**